lrpar/src/lib/ctbuilder.rs

```diff
@@ -363,24 +365,19 @@ where
                         // Iterate over all $-arguments and replace them with their respective
                         // element from the argument vector (e.g. $1 is replaced by args[0]). At
                         // the same time extract &str from tokens and actiontype from nonterminals.
-                        outs.push_str(&format!("fn {prefix}action_{}({prefix}input: &str, {prefix}args: &[AStackType<{actiont}, {}>]) -> {actiont} {{
+                        outs.push_str(&format!("fn {prefix}action_{}(_ridx: RIdx<{storaget}>, {prefix}input: &str, mut {prefix}args: vec::Drain<AStackType<{actiont}, {storaget}>>) -> {actiont} {{
 ",
                             usize::from(pidx),
-                            StorageT::type_name(),
+                            storaget=StorageT::type_name(),
                             prefix=ACTION_PREFIX,
                             actiont=actiontype));
-                        for m in re.find_iter(s) {
-                            let num = match &s[m.start() + 1..m.end()].parse::<usize>() {
-                                Ok(val) => val - 1,
-                                Err(_) => unreachable!()
-                            };
+                        for i in 0..grm.prod(pidx).len() {
                             outs.push_str(&format!(
-                                "    let {prefix}arg_{} = match {prefix}args[{}] {{",
-                                num + 1,
-                                num,
+                                "    let {prefix}arg_{} = match {prefix}args.next().unwrap() {{",
```

Comment:
I like this! The drain is nicer than the reusable Vector.

Reply:
One question though. Doesn't that generate variables that might not be used, creating `unused variable` warnings when compiling things?